### ISSUE TITLE
Integrate "live-reload port" into 4200

### ISSUE THREAD

**Issue by Turbo87** [ember-cli/ember-cli#6913]
In the current state Ember CLI essentially opens two HTTP servers. One for the assets and one to support live reloading on file changes. We should try to figure out if it is possible to integrate the second server into the first one hidden behind some private path so that we only have to open a single port. This would also work around the #6513 issue.

@rwjblue mentioned that he already started working on this

**Comment by kolybasov** [comment]
Hi @Turbo87 ! Looks like this issue don't have any progress for now. I can grab this one and try to refactor livereload-server task to livereload-middleware. [tiny-lr](https://github.com/mklabs/tiny-lr#middleware) supports it's usage as middleware so probably this will be not tricky. Let me know if this issue is actual.

**Comment by Turbo87** [comment]
@kolybasov yeah, sounds good!

**Comment by stefanpenner** [comment]
ya it shouldn't be tricky. I always intended on refactoring to this approach, but never got around to it.

👍

**Comment by kolybasov** [comment]
@stefanpenner can I create `lib/tasks/server/middleware/live-reload/index.js` addon and implement class:

```javascript
class LiveReloadAddon {
  shouldAddMiddleware() {}
  serverMiddleware() {}
}
```

or I have chosen wrong thing to use?

**Comment by stefanpenner** [comment]
Fixed by https://github.com/ember-cli/ember-cli/pull/7940 which is available in 3.4.0-beta.2
